executor/runtime/docker/docker.go

```diff
@@ -499,6 +511,22 @@ func (r *DockerRuntime) dockerConfig(c *runtimeTypes.Container, binds []string,
 	return containerCfg, hostCfg, nil
 }
 
+// computeHostname computes a hostname in the container using container ID or ec2 style
+// depending on titusParameter.agent.hostnameStyle setting.  Return error if style is unrecognized.
+func computeHostname(c *runtimeTypes.Container) (string, error) {
+	hostnameStyle := strings.ToLower(c.TitusInfo.GetPassthroughAttributes()[hostnameStyleParam])
+	switch hostnameStyle {
+	case defaultStyle:
+		return strings.ToLower(c.TaskID), nil
+	case ec2Style:
+		hostname := fmt.Sprintf("ip-%s", strings.Replace(c.Allocation.IPV4Address, ".", "-", 3))
+		log.WithField(hostnameStyleParam, hostnameStyle).Infof("Hostname set to EC2 style: %s", hostname)
+		return hostname, nil
+	default:
+		return "", &runtimeTypes.InvalidConfigurationError{Reason: fmt.Errorf("Unknown hostname style: %s", hostnameStyle)}
```

Comment:
Why not make invalid hostname its own error type?

Reply:
I was think that InvalidConfigurationError is reusable for other future configuration error.  Creating a InvalidHostnameStyleError is a bit too narrow.